### PR TITLE
[#366] Add wait for container removed

### LINKED ISSUE #366: Bucket removal can behave incorrectly
In current implementation bucket removal produces container removal https://github.com/nspcc-dev/neofs-s3-gw/blob/44548212858db1d13a27c9b89cfe55041cde1e00/api/layer/container.go#L241.
This operation is not synchronous. I didn't catch any implementation requirements in docs, so may be we need to make sync operation and await for the actual removal (like in bucket creation).

### PR DESCRIPTION
**Pull request by KirillovDenis**
closes #366

### CODE REVIEW

File: api/layer/container.go [hunk 1]
```diff
@@ -207,6 +207,43 @@ func (n *layer) waitEACLPresence(ctx context.Context, cid *cid.ID, table *eacl.T
...
+		return err
+	}
+
+	return n.waitForContainerRemoved(ctx, cid, defaultWaitParams())
+}
+
+func (n *layer) waitForContainerRemoved(ctx context.Context, cid *cid.ID, params *waitParams) error {
+	return waitFor(ctx, params, func(ctx context.Context) bool {
+		_, err := n.pool.GetContainer(ctx, cid)
+		if err != nil && strings.Contains(err.Error(), "container not found") {
```
Comment (cthulhu-rider) [inline api/layer/container.go:239]: Lets add `TODO: (neofs-s3-gw#367) handle NeoFS API status error`?

Also we can just `return` this statement w/o `if` block.

File: api/layer/container.go [hunk 2]
```diff
@@ -207,6 +207,43 @@ func (n *layer) waitEACLPresence(ctx context.Context, cid *cid.ID, table *eacl.T
...
+	}
+
+	if err = n.pool.DeleteContainer(ctx, cid, pool.WithSession(sessionToken)); err != nil {
+		return err
+	}
+
+	return n.waitForContainerRemoved(ctx, cid, defaultWaitParams())
+}
+
+func (n *layer) waitForContainerRemoved(ctx context.Context, cid *cid.ID, params *waitParams) error {
```
Comment (cthulhu-rider) [inline api/layer/container.go:236]: Minor suggestion: rename `cid` parameter in order to no collide with package name (in `deleteContainer` too).

File: api/layer/container.go [hunk 3]
```diff
@@ -207,6 +207,43 @@ func (n *layer) waitEACLPresence(ctx context.Context, cid *cid.ID, table *eacl.T
...
+	return waitFor(ctx, params, func(ctx context.Context) bool {
+		_, err := n.pool.GetContainer(ctx, cid)
+		if err != nil && strings.Contains(err.Error(), "container not found") {
+			return true
+		}
+		return false
+	})
+}
+
+func waitFor(ctx context.Context, params *waitParams, fn func(context.Context) bool) error {
```
Comment (cthulhu-rider) [inline api/layer/container.go:246]: Lets clarify in docs that it `waits for fn's true return`?

File: api/layer/container.go (hunk 1 above)
Comment (KirillovDenis) [reply api/layer/container.go:232]: Currently, it doesn't work. Do we need use neofs-node version other than v0.27.5?

Comment (cthulhu-rider) [reply api/layer/container.go:232]: I know, meant just to add the TODO comment.

**Pull request merged**